ga4gh/client.py

```diff
@@ -38,31 +145,53 @@ def runRequest(self, request, url, protocolClass, listAttr):
         """
         notDone = True
         while notDone:
-            jsonString = request.toJSONString()
-            headers = {"Content-type": "application/json"}
-            # make sure we correctly join with/out trailing slashes
-            fullUrl = posixpath.join(self._urlPrefix, url)
-            # TODO Can we get requests to output debugging information when
-            # debugLevel > 0?
-            authUrl = self._addAuth(fullUrl)
-            response = requests.post(
-                authUrl, jsonString, headers=headers, verify=False)
-            response.raise_for_status()
-            jsonString = response.text
-            self._bytesRead += len(jsonString)
-            if self._debugLevel > 1:
-                # TODO use a logging output and integrate with HTTP client more
-                # nicely.
-                print("json response:")
-                pp = json.dumps(
-                    json.loads(jsonString), sort_keys=True, indent=4)
-                print(pp)
-            resp = protocolClass.fromJSONString(jsonString)
-            # TODO handle HTTP errors from requests and display.
-            for extract in getattr(resp, listAttr):
-                yield extract
-            request.pageToken = resp.nextPageToken
-            notDone = resp.nextPageToken is not None
+            # do the request
+            response = self._doRequest(requestContainer)
+
+            # deserialize json response
+            responseObject = self._deserializeResponse(
+                response, requestContainer.protocolClass)
+
+            # yield the results
+            if requestContainer.listAttr is None:
+                yield responseObject
+            else:
+                obj = getattr(responseObject, requestContainer.listAttr)
+                for extract in obj:
+                    yield extract
+
+            # update the pageToken and notDone
+            notDone = self._updateNotDone(responseObject, requestContainer)
+
+    def getReferenceSet(self, id_):
+        """
+        Returns a referenceSet from the server
+        """
+        requestContainer = self.RunRequestContainer(
+            url="referencesets/{id}",
+            protocolClass=protocol.GAReferenceSet,
+            requestMethod='get', id_=id_)
+        return self.runRequestSingle(requestContainer)
```

Comment:
The point of the RunRequestContainer was to decouple the cli methods from the client methods by providing an object that any cli method could create and any client method could use.  I'm not sure tightly coupling them again would be a good idea.